apps/dashboard/app/views/layouts/application.html.erb

```diff
@@ -43,25 +43,19 @@
             <%= render partial: 'layouts/nav/sessions', nav_groups: @nav_groups if Configuration.app_development_enabled? || @nav_groups.any?(&:has_batch_connect_apps?) %>
             <%= render partial: 'layouts/nav/all_apps' if @user_configuration.show_all_apps_link %>
           <% else %>
-            <% @nav_bar.each do |nav_item| %>
-              <%= render partial: nav_item.partial_path, locals: nav_item.to_h %>
-            <% end %>
+            <%= render partial: "layouts/nav/custom_navigation", locals: { navigation: @nav_bar }%>
           <% end %>
         </ul>
 
         <ul class="navbar-nav">
-          <%= render partial: 'layouts/nav/develop_dropdown' if Configuration.app_development_enabled? %>
-          <%= render partial: "layouts/nav/help_dropdown" %>
-
-          <li class="nav-item" data-container="body" data-toggle="popover" data-content="<%= t('dashboard.nav_user', username: @user.name) %>" data-placement="bottom">
-            <a class="nav-link disabled">
-              <i class="fas fa-user" aria-hidden="true" title="<%= t('dashboard.nav_user', username: @user.name) %>" aria-hidden="true"></i><span class="d-sm-none d-md-none d-lg-inline"> <%= t('dashboard.nav_user', username: @user.name) %></span>
-            </a>
-          </li>
-
-          <li class="nav-item">
-            <a class="nav-link" href="/logout"><i class="fas fa-sign-out-alt" aria-hidden="true"></i><span class="d-sm-none d-md-none d-lg-inline"> <%= t('dashboard.nav_logout') %></span></a>
-          </li>
+          <% if @help_bar.empty? %>
+            <%= render partial: 'layouts/nav/develop_dropdown' if Configuration.app_development_enabled? %>
+            <%= render partial: "layouts/nav/help_dropdown" %>
+            <%= render partial: "layouts/nav/user" %>
+            <%= render partial: "layouts/nav/log_out" %>
+          <% else %>
+            <%= render partial: "layouts/nav/custom_navigation", locals: { navigation: @help_bar }%>
```

Comment:
Can you remove `custom_navigation.html.erb` and have this follow the same logic as the other one above? It's just easier when they both do the same thing.

Reply:
Oh, nevermind I see you have update the lines above.